obsah/__init__.py

```diff
@@ -36,6 +36,51 @@
 Variable = namedtuple('Variable', ['name', 'parameter', 'help_text', 'action', 'type', 'choices'])
 
 
+class AppendUniqueAction(argparse.Action):
+    """
+    Custom argparse action that appends values but ensures uniqueness.
+    Similar to action='append', but duplicate values are ignored.
+    """
+    def __init__(self, option_strings, dest, default=None, **kwargs):
+        if default is None:
+            default = []
+        super().__init__(option_strings, dest, default=default, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        items = getattr(namespace, self.dest, None)
+        if items is None:
+            items = []
+        # Create a new list to avoid modifying the default
+        items = list(items)
+        # Only append if the value is not already in the list
+        if values not in items:
+            items.append(values)
+        setattr(namespace, self.dest, items)
+
+
+class RemoveAction(argparse.Action):
+    """
+    Custom argparse action that removes values from a list.
+    Useful when you have a default list and want to selectively remove items.
+    """
+    def __init__(self, option_strings, dest, default=None, **kwargs):
+        if default is None:
+            default = []
+        super().__init__(option_strings, dest, default=default, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        items = getattr(namespace, self.dest, None)
+        if items is None:
+            items = []
+        else:
+            # Create a new list to avoid modifying the default
+            items = list(items)
+            # Remove the value if it exists in the list
+            if values in items:
+                items.remove(values)
+        setattr(namespace, self.dest, items)
```

Comment:
This makes me question: if `items` is `None`, should this whole action be a noop?

Reply:
Maybe, but I also don't think it really hurts?